components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridConfigPersistenceManagerTest.java

```diff
@@ -0,0 +1,136 @@
+@WithRealmService(injectToSingletons = {EntitlementConfigHolder.class}, initUserStoreManager = true)
+@WithH2Database(jndiName = "jdbc/WSO2IdentityDB", files = {"dbscripts/h2.sql"})
+public class HybridConfigPersistenceManagerTest {
+
+    private HybridConfigPersistenceManager hybridConfigPersistenceManager;
+    private RegistryConfigPersistenceManager registryConfigPersistenceManager;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        hybridConfigPersistenceManager = new HybridConfigPersistenceManager();
+        registryConfigPersistenceManager = new RegistryConfigPersistenceManager();
+    }
+
+    @Test
+    public void testGetDefaultGlobalPolicyAlgorithmName() throws Exception {
+
+        String globalPolicyAlgorithmName = registryConfigPersistenceManager.getGlobalPolicyAlgorithmName();
+        assertEquals(globalPolicyAlgorithmName, DENY_OVERRIDES);
+    }
+
+    @DataProvider
+    public Object[][] globalPolicyAlgorithmData() {
+
+        return new Object[][]{
+                {DENY_OVERRIDES},
+                {PERMIT_OVERRIDES},
+                {FIRST_APPLICABLE},
+                {ONLY_ONE_APPLICABLE},
+                {ORDERED_DENY_OVERRIDES},
+                {ORDERED_PERMIT_OVERRIDES}
+        };
+    }
+
+    @Test(dataProvider = "globalPolicyAlgorithmData")
+    public void testGetGlobalPolicyAlgorithmNameFromRegistry(String policyAlgorithm) throws Exception {
+
+        // Add the global policy combining algorithm.
+        boolean addResult = registryConfigPersistenceManager.addOrUpdateGlobalPolicyAlgorithm(policyAlgorithm);
+        assertFalse(addResult);
+
+        String policyAlgorithmFromRegistry = hybridConfigPersistenceManager.getGlobalPolicyAlgorithmName();
+        assertEquals(policyAlgorithmFromRegistry, policyAlgorithm);
+
+        registryConfigPersistenceManager.deleteGlobalPolicyAlgorithm();
+    }
+
+    @Test(dataProvider = "globalPolicyAlgorithmData")
+    public void testGetGlobalPolicyAlgorithmNameFromDatabase(String policyAlgorithm) throws Exception {
+
+        // Add the global policy combining algorithm.
+        boolean addResult = hybridConfigPersistenceManager.addOrUpdateGlobalPolicyAlgorithm(policyAlgorithm);
+        assertFalse(addResult);
+
+        String policyAlgorithmFromDatabase = hybridConfigPersistenceManager.getGlobalPolicyAlgorithmName();
+        assertEquals(policyAlgorithmFromDatabase, policyAlgorithm);
+
+        deletePolicyCombiningAlgorithmInDatabase();
+    }
+
+    @Test(dataProvider = "globalPolicyAlgorithmData")
+    public void testUpdateGlobalPolicyAlgorithm(String policyAlgorithm) throws Exception {
+
+        // Add the global policy combining algorithm.
+        boolean addResult = hybridConfigPersistenceManager.addOrUpdateGlobalPolicyAlgorithm(DENY_OVERRIDES);
+        assertFalse(addResult);
+
+        boolean updateResult = hybridConfigPersistenceManager.addOrUpdateGlobalPolicyAlgorithm(policyAlgorithm);
+        assertTrue(updateResult);
```

Comment:
This validates the data, lets improve the test to validate the logic.
ie. This test does not gurentee that we have deleted the record in the registry which is a crucial part of ondemand migration logic this HybridPersistent manager is responsible of.

Reply:
Fixed in the latest changes